syntax = "proto3";
package collector;




import "inspection_sdk/model/inspection/arg.proto";

/*
CreateCollector请求
*/
message CreateCollectorRequest {
    /*
    套件id
    */
    string pluginId = 1 ;
    /*
    脚本名称
    */
    string name = 2 ;
    /*
    脚本内容
    */
    string content = 3 ;
    /*
    脚本类型
    */
    string script = 4 ;
    /*
    脚本入参
    */
    repeated inspection.InspectionArg args = 5 ;
}

/*
CreateCollector返回
*/
message CreateCollectorResponse {
    /*
    采集脚本Id
    */
    string collectorId = 1 ;
}

/*
CreateCollectorApi返回
*/
message CreateCollectorResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    collector.CreateCollectorResponse data = 4 ;
}

